File: 프로그래머스/lv2/76502. 괄호 회전하기/괄호 회전하기.java
import java.util.*;

class Solution {
    public int solution(String s) {
        int answer = 0;
        int len = s.length();

        for (int i = 0; i < len; i++) {
            Stack<Character> stack = new Stack<>();
            String str = s.substring(i, s.length()) + s.substring(0, i);

            for (int j = 0; j < str.length(); j++) {
                char ch = str.charAt(j);
                if (stack.isEmpty())
                    stack.push(ch);
                else if (ch == ')' && stack.peek() == '(')
                    stack.pop();
                else if (ch == '}' && stack.peek() == '{')
                    stack.pop();
                else if (ch == ']' && stack.peek() == '[')
                    stack.pop();
                else
                    stack.push(ch);
            }

            if (stack.isEmpty())
                answer++;
        }

        return answer;
    }
}

File: 프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.py
def solution(s):
    answer = 0
    for i in range(len(s)) :
        arr = []
        target = s[i:] + s[:i]
        for j in target :
            if len(arr) == 0 :
                arr.append(j)
            elif j == ')' and arr[-1] == '(' :
                arr.pop()
            elif j == ']' and arr[-1] == '[' :
                arr.pop()
            elif j == '}' and arr[-1] == '{' :
                arr.pop()
            else :
                arr.append(j)
        if len(arr) == 0 :
            answer += 1
                
    return answer